File: sga_backend/app/models/usuario.py
from sqlalchemy import (
    Column,
    String,
    Boolean,
    TIMESTAMP,
    text
)

from app.config.database import Base


class Usuario(Base):

    __tablename__ = "usuario"

    id_usuario = Column(
        String(20),
        primary_key=True
    )

    rol_usuario = Column(
        String(20),
        nullable=False,
        server_default="cliente"
    )

    nombres_usuario = Column(
        String(100),
        nullable=False
    )

    apellidos_usuario = Column(
        String(100),
        nullable=False
    )

    email_usuario = Column(
        String(100),
        unique=True
    )

    telefono_usuario = Column(
        String(20),
        nullable=False,
        unique=True
    )

    contrasena_usuario = Column(
        String(255)
    )

    tipo_documento = Column(
        String(20),
        nullable=False
    )

    fecha_creacion = Column(
        TIMESTAMP,
        nullable=False,
        server_default=text("CURRENT_TIMESTAMP")
    )

    fecha_actualizacion = Column(
        TIMESTAMP,
        nullable=False,
        server_default=text("CURRENT_TIMESTAMP")
    )

    estado_usuario = Column(
        Boolean,
        nullable=False,
        server_default="true"
    )

File: sga_frontend_jwt/src/interfaces/Usuario.ts
// Debe coincidir EXACTAMENTE con app/schemas/usuario_schema.py y
// app/controllers/usuario_controller.py del backend.

export type TipoDocumento = "CC" | "CE" | "NIT" | "PPT";

export type RolUsuario =
  | "admin"
  | "encargado_facturacion"
  | "encargado_logistico"
  | "cliente";

export interface Usuario {
  id_usuario: string;
  rol_usuario: RolUsuario;
  nombres_usuario: string;
  apellidos_usuario: string;
  email_usuario: string | null;
  telefono_usuario: string;
  tipo_documento: TipoDocumento;
  fecha_creacion: string;
  fecha_actualizacion?: string;
  estado_usuario: boolean;
}

export interface UsuarioCreatePayload {
  id_usuario: string;
  rol_usuario: RolUsuario;
  nombres_usuario: string;
  apellidos_usuario: string;
  email_usuario?: string | null;
  telefono_usuario: string;
  contrasena_usuario?: string | null;
  tipo_documento: TipoDocumento;
}

export interface LoginPayload {
  telefono_usuario?: string;
  email_usuario?: string;
  contrasena_usuario: string;
}

// Respuesta de POST /usuarios/login (el backend NO emite JWT; ver README).
export interface LoginResponse {
  logueado: boolean;
  id_usuario: string;
  rol_usuario: RolUsuario;
  nombres_usuario: string;
  apellidos_usuario: string;
  // NUEVO en la variante con JWT (no existe en la variante sin JWT)
  access_token: string;
  token_type: string;
}
